test/constructors.jl

```diff
@@ -128,4 +132,23 @@ include("common.jl")
             @test AdvancedHMC.getadaptor(state) isa expected.adaptor_type
         end
     end
+
+    @testset "Kernel hyperparameters of $(nameof(typeof(sampler)))" for (
+        sampler,
+        expected,
+    ) in [
+        (HMC(25), (kernel_hp = [25],)),
+        (HMCDA(0.8, 2.0), (kernel_hp = [2.0],)),
+        (NUTS(0.8; max_depth = 20, Δ_max = 2000.0), (kernel_hp = [20, 2000.0],)),
+    ]
+
+        # Step.
+        rng = Random.default_rng()
+        transition, state =
+            AbstractMCMC.step(rng, model, sampler; n_adapts = 0, init_params = θ_init)
+
+        # Verify that the state is what we expect.
+
+        @test get_kernel_hyperparams(sampler, state) == expected.kernel_hp
+    end
```

Comment:
I'd recommend just putting this with the other testset so we get the same test coverage.

Also, can you add tests for the elementypes of the parameters?

Reply:
Done!